torch/utils/cpp_extension.py

```diff
@@ -740,6 +753,16 @@ def _check_abi(self):
                    'Please set `DISTUTILS_USE_SDK=1` and try again.')
             raise UserWarning(msg)
 
+    def _check_cuda_version(self):
+        if CUDA_HOME:
+            nvcc = os.path.join(CUDA_HOME, 'bin', 'nvcc')
+            cuda_version_str = subprocess.check_output([nvcc, '--version']).strip().decode()
+            cuda_version = re.search(r'release (\d+[.]\d+)', cuda_version_str)
+            if cuda_version is not None:
+                cuda_version = cuda_version.group(1)
+                if cuda_version != torch.version.cuda:
```

Comment:
Shouldn't this assert if:
1. cuda is None - you can't build CUDA extensions w/o system-wide cuda
2. major version mismatches?

Reply:
Thanks for the quick review @stas00! 

1. I should verify if there is at least one CUDA extension to compile before checking if the warning should appear or not, otherwise it won't make much sense when compiling a C++ only extension.
2. True, this check should be more detailed